src/OrbitGl/TrackHeader.cpp

```diff
@@ -0,0 +1,151 @@
+    // defined by TrackContentTopMargin and TrackContentBottomMargin.
+    // Both margins are factored into the total height of each track.
+    Vec2 top_left(indentation_x0, y0);
+    Vec2 tab_top_right(top_left[0] + label_width, top_left[1]);
+    Vec2 tab_bottom_right(top_left[0] + label_width, top_left[1] + label_height);
+
+    auto shared_this = shared_from_this();
+    DrawTriangleFan(batcher, rounded_corner, top_left, GlCanvas::kBackgroundColor, 90.f, track_z,
+                    shared_this);
+    DrawTriangleFan(batcher, rounded_corner, tab_top_right, GlCanvas::kBackgroundColor, 180.f,
+                    track_z, shared_this);
+    DrawTriangleFan(batcher, rounded_corner, tab_bottom_right, track_background_color, 0, track_z,
+                    shared_this);
+  }
+
+  // Draw label.
+  if (!picking) {
+    uint32_t font_size = layout_->CalculateZoomedFontSize();
+    // For the first 5 indentations, we decrease the font_size by 10 percent points (per
+    // indentation).
+    constexpr uint32_t kMaxIndentationLevel = 5;
+    uint32_t capped_indentation_level =
+        std::min(track->GetIndentationLevel(), kMaxIndentationLevel);
+    font_size = (font_size * (10 - capped_indentation_level)) / 10;
+    float label_offset_x = layout_->GetTrackLabelOffsetX();
+
+    const Color kColor =
+        track->IsTrackSelected() ? GlCanvas::kTabTextColorSelected : Color(255, 255, 255, 255);
+
+    TextRenderer::TextFormatting formatting{font_size, kColor, label_width - label_offset_x};
+    formatting.valign = TextRenderer::VAlign::Middle;
+
+    text_renderer.AddTextTrailingCharsPrioritized(
+        track->GetLabel().c_str(), indentation_x0 + label_offset_x, y0 + half_label_height, text_z,
+        formatting, track->GetNumberOfPrioritizedTrailingCharacters());
+  }
+}
+
+void orbit_gl::TrackHeader::DoUpdateLayout() {
+  CaptureViewElement::DoUpdateLayout();
+
+  UpdatePositionOfCollapseToggle();
+}
+
+void TrackHeader::UpdatePositionOfCollapseToggle() {
+  Track* track = dynamic_cast<Track*>(GetParent());
+  ORBIT_CHECK(track != nullptr);
+
+  const float label_height = layout_->GetTrackTabHeight();
+  const float half_label_height = 0.5f * label_height;
+  const float x0 = GetPos()[0] + layout_->GetTrackTabOffset() +
+                   layout_->GetTrackIndentOffset() * track->GetIndentationLevel();
+  const float button_offset = layout_->GetCollapseButtonOffset();
+  const float toggle_y_pos = GetPos()[1] + half_label_height;
+  Vec2 toggle_pos = Vec2(x0 + button_offset, toggle_y_pos);
+
+  float size = layout_->GetCollapseButtonSize(track->GetIndentationLevel());
+  collapse_toggle_->SetWidth(size);
+  collapse_toggle_->SetHeight(size);
+  collapse_toggle_->SetPos(toggle_pos[0], toggle_pos[1]);
+
+  collapse_toggle_->SetIsCollapsible(track->IsCollapsible());
+}
+
+void TrackHeader::OnDrag(int x, int y) {
+  CaptureViewElement::OnDrag(x, y);
+
+  GetParent()->SetPos(GetPos()[0], mouse_pos_cur_[1] - picking_offset_[1]);
```

Comment:
I think it's the only place we do this, yes. Long-term plan is to implement this similar to the scrollwheel, but I'm leaning towards leaving the handling of dragging in the child still, because it seems more intuitive to only have the tab as a draggable element instead of the whole track background.

Reply:
I've changed this now to use an explicit `DragTrack` method from the new `TrackControlInterface`.